ReactNativeClient/lib/BaseApplication.js

```diff
@@ -332,6 +346,17 @@ class BaseApplication {
 		}, 1000);
 	}
 
+	async waitForAutoAddResourcesFinished() {
+		return new Promise((resolve) => {
+			const iid = setInterval(() => {
+				if (!ResourceFetcher.instance().autoAddResourcesCalls_.length) {
+					clearInterval(iid);
+					resolve();
+				}
+			}, 100);
+		});
+	}
```

Comment:
> Could that be part of ResourceFetcher?

Yes, it should be.  I was keeping it close to the `scheduleAutoAddResources()` which has the timer we are trying to stop.  I think they both probably belong in the `ResourceFetcher`.

> ResourceFetcher::waitForAllFinished() does?

Similar but not the same.  `waitForAllFinished()` is waiting on a broader set of conditions, where this method is specifically waiting just for the timer that is run in `BaseApplication`.

If we move `scheduleAutoAddResources()` to `ResourceFetcher` from `BaseApplication`, then I think it can be cleaned up better.  Would that be ok?

Reply:
Ok, I've moved it to `ResourceFetcher` and appears to fall into place nicely.

Be aware: this change involves a refactor of app code and not just code that runs during test.  I can revert if you're not comfortable with that.